include/csound_files.h

```diff
@@ -176,6 +207,19 @@ extern "C" {
   PUBLIC void csoundSetFileOpenCallback(CSOUND *p,
                                         void (*func)(CSOUND*, const char*,
                                                      int32_t, int32_t, int32_t));
+
+  /** 
+   * Sets external callbacks for the soundfile interface
+   * replacing any default callbacks. These functions are used by
+   * Csound to perform soundfile IO throughout the system.
+   * These are set to sndfile.h functions if USE_LIBSNDFILE is defined
+   * otherwise they are set to non-op stubs.
+   * Callbacks are passed via the SNDFILE_CALLBACKS struct.
+   * Any NULL callback pointer will disable setting of the specific callback
+   * If p is NULL then all callbacks are reverted to default values.
+   **/
+  PUBLIC void csoundSetSndfileCallbacks(CSOUND *csound, SNDFILE_CALLBACKS *p);
```

Comment:
This makes me think we might follow a similar pattern for graph callbacks rather than individual API calls.